utils/ixdtf/src/error.rs

```diff
@@ -0,0 +1,123 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! An error enum for representing `ixdtf` parsing errors.
+
+#[non_exhaustive]
+#[derive(PartialEq, Clone, Copy, Debug)]
+pub enum ParserError {
+    ParseFloat,
+    AbruptEnd,
+    InvalidEnd,
+    InvalidMonthRange,
+    InvalidDayRange,
+    InvalidYearRange,
+    DateYear,
+    DateExtendedYear,
+    DateFourDigitYear,
+    DateMonth,
+    DateDay,
+    DateUnexpectedEnd,
+    TimeHour,
+    TimeMinute,
+    TimeSecond,
+    FractionPart,
+    DateSeparator,
+    TimeSeparator,
+    DecimalSeparator,
+
+    // Missing Required components.
+    MissingRequiredTzAnnotation,
+    MissingRequiredTime,
+    MissingUtcOffset,
+
+    // Annotation Related Errors
+    InvalidAnnotation,
+    AnnotationOpen,
+    AnnotationClose,
+    AnnotationChar,
+    AnnotationKeyValueSeparator,
+    AnnotationKeyLeadingChar,
+    AnnotationKeyChar,
+    AnnotationValueCharPostHyphen,
+    AnnotationValueChar,
+    UnrecognizedCritical,
+    CriticalDuplicateCalendar,
+
+    // Time Zone Errors
+    TzLeadingChar,
+    IanaCharPostSeparator,
+    IanaChar,
+    UtcTimeSeparator,
+
+    // Duration Errors
+    DurationDisgnator,
+    DateDurationPartOrder,
+    TimeDurationPartOrder,
+    TimeDurationDesignator,
+}
+
+impl ParserError {
+    pub(crate) fn abrupt_end() -> Self {
+        ParserError::AbruptEnd
+    }
+}
+
+impl core::fmt::Display for ParserError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        let msg = match self {
+            ParserError::ParseFloat => "Invalid float while parsing fraction part.",
```

Comment:
Suggestion: use `displaydoc` to put these display strings inline with the enum definitions. We use this dependency widely in icu4x code.